src/garage/torch/algos/maml/maml.py

```diff
@@ -0,0 +1,478 @@
+                               itr,
+                               all_samples,
+                               all_params,
+                               set_grad=True):
+        """Compute KL divergence.
+
+        For each task, compute the KL divergence between the old policy
+        distribution and current policy distribution.
+
+        Args:
+            itr (int): Iteration number.
+            all_samples (list[list[MAMLTrajectoryBatch]]): Two
+                dimensional list of MAMLTrajectoryBatch of size
+                [meta_batch_size * (num_grad_updates + 1)]
+            all_params (list[dict]): A list of named parameter dictionaries.
+                Each dictionary contains key value pair of names (str) and
+                parameters (torch.Tensor).
+            set_grad (bool): Whether to enable gradient calculation or not.
+
+        Returns:
+            torch.Tensor: Calculated mean value of KL divergence.
+
+        """
+        theta = dict(self._policy.named_parameters())
+        old_theta = dict(self._old_policy.named_parameters())
+
+        kls = []
+        for task_samples, task_params in zip(all_samples, all_params):
+            for i in range(self._num_grad_updates):
+                self._adapt(itr, task_samples[i], set_grad=set_grad)
+
+            update_module_params(self._old_policy, task_params)
+            with torch.set_grad_enabled(set_grad):
+                # pylint: disable=protected-access
+                kl = self._inner_algo._compute_kl_constraint(
+                    task_samples[-1].observations)
+            kls.append(kl)
+
+            update_module_params(self._policy, theta)
+            update_module_params(self._old_policy, old_theta)
+
+        return torch.stack(kls).mean()
+
+    def _compute_loss(self, itr, batch_samples):
+        """Compute loss for one task and one gradient step.
+
+        Args:
+            itr (int): Iteration number.
+            batch_samples (MAMLTrajectoryBatch): Samples data for one
+                task and one gradient step.
+
+        Returns:
+            torch.Tensor: Computed loss value.
+
+        """
+        # pylint: disable=protected-access
+        return self._inner_algo._compute_loss(itr, batch_samples.observations,
+                                              batch_samples.actions,
+                                              batch_samples.rewards,
+                                              batch_samples.lengths,
+                                              batch_samples.baselines)
+
+    def _compute_policy_entropy(self, task_samples):
+        """Compute policy entropy.
+
+        Args:
+            task_samples (list[MAMLTrajectoryBatch]): Samples data for
+                one task.
+
+        Returns:
+            torch.Tensor: Computed entropy value.
+
+        """
+        # pylint: disable=protected-access
+        entropies = [
+            self._inner_algo._compute_policy_entropy(samples.observations)
```

Comment:
why not stack the samples first, then compute the entropies vectorized? shouldn't that be faster?